src/Symfony/Component/Serializer/Dumper/NormalizerDumper.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Dumper;
+
+use Symfony\Component\Serializer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+
+/**
+ * @author Guilhem Niot <guilhem.niot@gmail.com>
+ */
+final class NormalizerDumper
+{
+    private $classMetadataFactory;
+
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory)
+    {
+        $this->classMetadataFactory = $classMetadataFactory;
+    }
+
+    public function dump($class, array $context = array())
+    {
+        $reflectionClass = new \ReflectionClass($class);
+        if (!isset($context['class'])) {
+            $context['class'] = $reflectionClass->getShortName().'Normalizer';
```

Comment:
I really think we should introduce a `ClassNameConverter` here, I made one on my fork that's just doing `str_replace('\\', '_', $reflectionClass->getName())` to avoid collisions.